### PR TITLE
gh-91217: deprecate-sndhdr

### PR DESCRIPTION
**Pull request by brettcannon**
Automerge-Triggered-By: GH:brettcannon

### CODE REVIEW

File: Misc/NEWS.d/next/Library/2022-04-17-12-27-46.gh-issue-91217.tNDWtK.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Deprecate the sndhdr module, as well as inline needed functionality for
+email.mime.MIMEAudio .
```
Comment (hugovk) [inline Misc/NEWS.d/next/Library/2022-04-17-12-27-46.gh-issue-91217.tNDWtK.rst]:
```suggestion
``email.mime.MIMEAudio``.
```

File: Lib/email/mime/audio.py [hunk 2]
```diff
@@ -6,19 +6,43 @@
...
+
+def _test_wav(h, f):
+    import wave
+    # 'RIFF' <len> 'WAVE' 'fmt ' <len>
+    if not h.startswith(b'RIFF') or h[8:12] != b'WAVE' or h[12:16] != b'fmt ':
+        return None
+    else:
+        return "x-wav"
+
+_tests.append(_test_wav)
```
Comment (hugovk) [inline Lib/email/mime/audio.py:44]: PEP 8:

```suggestion

def _test_aifc_aiff(h, f):
    if not h.startswith(b'FORM'):
        return None
    if h[8:12] in {b'AIFC', b'AIFF'}:
        return 'x-aiff'
    else:
        return None


_tests.append(_test_aifc_aiff)


def _test_au(h, f):
    if h.startswith(b'.snd'):
        return 'basic'
    else:
        return None


_tests.append(_test_au)


def _test_wav(h, f):
    import wave
    # 'RIFF' <len> 'WAVE' 'fmt ' <len>
    if not h.startswith(b'RIFF') or h[8:12] != b'WAVE' or h[12:16] != b'fmt ':
        return None
    else:
        return "x-wav"


_tests.append(_test_wav)
```

File: Lib/email/mime/audio.py [hunk 3]
```diff
@@ -31,14 +55,14 @@ def _whatsnd(data):
     """
     hdr = data[:512]
     fakefile = BytesIO(hdr)
-    for testfn in sndhdr.tests:
+    for testfn in _tests:
         res = testfn(hdr, fakefile)
```
Comment (hugovk) [inline Lib/email/mime/audio.py:59]: None of the inlined `test_*` functions use this second parameter `fakefile`, shall we remove it?

File: Lib/email/mime/audio.py (hunk 2 above)
Comment (brettcannon) [reply Lib/email/mime/audio.py:44]: While I appreciate the suggestion, I don't think it makes it any easier to read and actually moves the `append()` farther away from the actual code being stored.

File: Lib/email/mime/audio.py (hunk 3 above)
Comment (brettcannon) [reply Lib/email/mime/audio.py:59]: I'm assuming @warsaw is going to rewrite all of this code anyway, so I will leave it to him. 😁

Comment (warsaw) [reply Lib/email/mime/audio.py:59]: Ha!  Yes, I will take a look when I get a chance.

**Pull request merged**